src/Tutorials/Unit-Testing-Grains.md

```diff
@@ -5,81 +5,159 @@ title: Unit Testing Grains
 using Orleans;
-using Orleans.TestingHost;
 using Xunit;
+using Moq;
 
-namespace Tests
+namespace Tests 
 {
-    public class HelloWorldSiloTests : TestingSiloHost, IDisposable
+    public class WorkerGrainTests 
     {
-        public void Dispose()
+        [Fact]
+        public async Task RecordsMessageInJournal() 
         {
-            // Optional.
-            // By default, the next test class which uses TestingSiloHost will
-            // cause a fresh Orleans silo environment to be created.
-            StopAllSilos();
+            var data = "Hello, World";
+
+            var journal = new Mock<IJournalGrain>();            
+
+            var worker = new Mock<WorkerGrain>();
+            myGrain.Setup(x => x.GrainFactory.GetGrain<IJournalGrain>(It.IsAny<Guid>())).Returns(journal.Object);
+
+            await worker.DoWork(data)
+
+            journal.Verfiy(x => x.Record(data), Times.Once());
         }
+    }
+}
+```
+
+Here we create our grain under test, `WorkerGrain`, using Moq which means we can then override the behaviour of the `GrainFactory` so that it returns a mocked `IJournalGrain`. 
+We can then verify that our `WorkerGrain` interacts with the `IJournalGrain` as we expect.
+
+### Using TestCluster
+
+The `Microsoft.Orleans.TestingHost` NuGet package contains `TestCluster` which can be used to create an in-memory cluster, comprised of two silos by default, which can be used to test grains.
 
+```csharp
+using System;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.TestingHost;
+using Xunit;
+
+namespace Tests 
+{
+    public class HelloGrainTests 
+    {
         [Fact]
-        public async Task SayHelloRespondsCorrectly()
+        public async Task SaysHelloCorrectly()
         {
-            // The Orleans silo / client test environment is already set up at this point.
-
-            const long id = 0;
-            const string greeting = "Bonjour";
+            var cluster = new TestCluster();
+            cluster.Deploy();
 
-            IHello grain = GrainFactory.GetGrain<IHello>(id);
+            var hello = cluster.GrainFactory.GetGrain<IHelloGrain>(Guid.NewGuid());
+            var greeting = await hello.SayHello();
 
-            // This will create and call a Hello grain with specified 'id' in one of the test silos.
-            string reply = await grain.SayHello(greeting);
+            cluster.StopAllSilos();
 
-            Assert.NotNull(reply);
-            string expected = $"You said: '{greeting}', I say: Hello!";
-            Assert.AreEqual(expected, reply);
+            Assert.Equal("Hello, World", greeting);
         }
     }
-}   
+}
+```
 
+Due to the overhead of starting an in-memory cluster you may wish to create a `TestCluster` and reuse it among multiple test cases.
+For example this can be done using Xunit's class or collection fixtures (see [https://xunit.github.io/docs/shared-context.html](https://xunit.github.io/docs/shared-context.html) for more details).
```

Comment:
even thought namespace is `Xunit` due to .NET design guidelines, when talking about the test framework you should probably refer to it as `xUnit`.